library/bignum_core.h

```diff
@@ -74,6 +74,57 @@ size_t mbedtls_mpi_core_bitlen( const mbedtls_mpi_uint *A, size_t A_limbs );
 void mbedtls_mpi_core_bigendian_to_host( mbedtls_mpi_uint *A,
                                          size_t A_limbs );
 
+/**
+ * \brief   Perform a safe conditional copy of MPI which doesn't reveal whether
+ *          the condition was true or not.
+ *
+ * \param[out] X        The address of the first MPI. This must be initialized.
+ *                      Must have enough limbs to store the full value of \p A.
+ * \param[in]  A        The address of the second MPI. This must be initialized.
+ * \param      limbs    The number of limbs of \p A.
+ * \param      assign   The condition deciding whether to perform the
+ *                      assignment or not. Must be either 0 or 1:
+ *                      * \c 1: Perform the assignment `X = A`.
+ *                      * \c 0: Keep the original value of \p X.
+ *
+ * \note           This function avoids leaking any information about whether
+ *                 the assignment was done or not.
```

Comment:
(minor) do we want to say "whether the condition was true or not" or "whether the assignment was true or not" (i.e. we are inconsistent in the description at the top of this comment block vs here).